src/ApplicationInsights.Kubernetes/EntityExtensions/PodExtensions.cs

```diff
@@ -1,49 +1,43 @@
+    public static IEnumerable<V1ContainerStatus> GetAllContainerStatus(this V1Pod pod)
     {
-        public static ContainerStatus? GetContainerStatus(this K8sPod self, string containerId)
-            => string.IsNullOrEmpty(containerId) ? null // Special case when container id is an empty string,
-                : self.Status?.ContainerStatuses?.FirstOrDefault(
-                    // Notice, we are using partial matching because there could be prefix of container ids like: docker://b1bf9cd89b57ba86c20e17bfd474638110e489da784a5e388983294d94ae9fc4
-                    status => !string.IsNullOrEmpty(status.ContainerID) && status.ContainerID.IndexOf(containerId, StringComparison.OrdinalIgnoreCase) != -1);  
-
-        public static IEnumerable<ContainerStatus> GetAllContainerStatus(this K8sPod pod)
+        if (pod?.Status is null)
         {
-            if (pod?.Status is null)
-            {
-                return Enumerable.Empty<ContainerStatus>();
-            }
-            return pod.Status.ContainerStatuses.NullAsEmpty();
+            return Enumerable.Empty<V1ContainerStatus>();
         }
+        return pod.Status.ContainerStatuses;
+    }
 
-        /// <summary>
-        /// Gets the ReplicaSet for the current pod.
-        /// </summary>
-        /// <param name="self">The target pod.</param>
-        /// <param name="scope">List of replicas to search from.</param>
-        /// <returns>Returns the replicaSet of the pod. Returns null when the data doesn't exist.</returns>
-        public static K8sReplicaSet? GetMyReplicaSet(this K8sPod self, IEnumerable<K8sReplicaSet> scope)
+    /// <summary>
+    /// Gets the ReplicaSet for the current pod.
+    /// </summary>
+    /// <param name="self">The target pod.</param>
+    /// <param name="scope">List of replicas to search from.</param>
+    /// <returns>Returns the replicaSet of the pod. Returns null when the data doesn't exist.</returns>
+    public static V1ReplicaSet? GetMyReplicaSet(this V1Pod self, IEnumerable<V1ReplicaSet> scope)
+    {
+        V1OwnerReference? replicaRef = self.Metadata.OwnerReferences.FirstOrDefault(owner => string.Equals(owner.Kind, V1ReplicaSet.KubeKind, StringComparison.Ordinal));
+        if (replicaRef != null)
         {
-            OwnerReference? replicaRef = self.Metadata?.OwnerReferences?.FirstOrDefault(owner => owner.GetKind() != null && owner.GetKind() == typeof(K8sReplicaSet));
-            if (replicaRef != null)
-            {
-                K8sReplicaSet? replica = scope?.FirstOrDefault(
-                    r => r.Metadata != null &&
-                    r.Metadata.Uid != null &&
-                    r.Metadata.Uid.Equals(replicaRef.Uid, StringComparison.OrdinalIgnoreCase));
-                return replica;
-            }
-            return null;
+            V1ReplicaSet? replica = scope?.FirstOrDefault(
+                r => string.Equals(r.Metadata.Uid, replicaRef.Uid, StringComparison.OrdinalIgnoreCase));
```

Comment:
It is interesting to find out, Uid is of type "GUID", and guid is case insensitive. :-)